sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/util/Utility.java

```diff
@@ -140,8 +141,12 @@ public static HttpPipeline buildHttpPipeline(ClientOptions clientOptions, HttpLo
         } else if (azureKeyCredential != null) {
             httpPipelinePolicies.add(new AzureKeyCredentialPolicy("api-key", azureKeyCredential));
         } else if (tokenCredential != null) {
-            httpPipelinePolicies.add(new BearerTokenAuthenticationPolicy(tokenCredential,
-                "https://search.azure.com/.default"));
+            if (audience == null) {
+                httpPipelinePolicies.add(new BearerTokenAuthenticationPolicy(tokenCredential,
+                    "https://search.azure.com/.default"));
```

Comment:
nit: if `audience` is null we should use `SearchAudience.AZURE_PUBLIC` instead of hardcoding the audience here.